specification/evaluation-context/evaluation-context.json

```
@@ -0,0 +1,23 @@
+[
+    {
+        "id": "Requirement 3.1",
+        "clean id": "requirement_3_1",
+        "content": "The `evaluation context` structure MUST define a required `targeting key` field of type string, identifying the subject of the flag evaluation.",
+        "RFC 2119 keyword": "MUST",
+        "children": []
+    },
+    {
+        "id": "Requirement 3.2",
+        "clean id": "requirement_3_2",
+        "content": "The evaluation context MUST define the following optional fields: `email` (string), `first name` (string), `last name`(string), `name`(string), `ip`(string), `tz`(string), `locale`(string), `country code` (string), `timestamp`(date), `environment`(string), `application`(string), and `version`(string).",
```

```suggestion
        "content": "The evaluation context MUST provide access the following optional attributes: `email` (string), `first name` (string), `last name`(string), `name`(string), `ip`(string), `tz`(string), `locale`(string), `country code` (string), `timestamp`(date), `environment`(string), `application`(string), and `version`(string).",
```

Optional suggestion: Making it clear that you want setters/getters for java, not raw field access.

---

> MUST provide access the following optional attributes

I see what you're getting at, but this suggest read-only to me (access, but not mutation).

---

I can get on board with that. I think the thing I actually care about is s/fields/attributes.

---

Where are all of these required optional fields coming from? I understand they're optional but why are their optionally enforced is it because of differences between all of the existing commercial providers?

---

At the risk of speaking for @justinabrahms , I think he's of the opinion we should definite these fields now. See my long explanation here: https://github.com/open-feature/spec/pull/60#discussion_r871587025

Let me know if this addresses your concern or not.

---

It appears to be a decent number of defined fields. I view it as we want to start with less of a surface API and build on it over time versus having a lot of fields up front that we don't know will be used and things are always harder to remove in the future then add.